lrpar/src/lib/ctbuilder.rs

```diff
@@ -541,48 +580,54 @@ where
         let mut actions: ::std::vec::Vec<&dyn Fn(::cfgrammar::RIdx<{storaget}>,
                        &'lexer dyn ::lrpar::NonStreamingLexer<'input, {storaget}>,
                        ::lrpar::Span,
-                       ::std::vec::Drain<::lrpar::parser::AStackType<{actionskind}<'input>, {storaget}>>)
-                    -> {actionskind}<'input>> = ::std::vec::Vec::new();\n",
+                       ::std::vec::Drain<::lrpar::parser::AStackType<{actionskind}<'input>, {storaget}>>,
+                       {param_typ})
+                    -> ({actionskind}<'input>, {param_typ})> = ::std::vec::Vec::new();\n",
                     actionskind = ACTIONS_KIND,
-                    storaget = type_name::<StorageT>()
+                    storaget = type_name::<StorageT>(),
+                    param_typ = self.param_type(grm),
                 ));
                 for pidx in grm.iter_pidxs() {
                     outs.push_str(&format!(
                         "        actions.push(&{prefix}wrapper_{});\n",
                         usize::from(pidx),
-                        prefix = ACTION_PREFIX
+                        prefix = ACTION_PREFIX,
                     ))
                 }
                 outs.push_str(&format!(
                     "
         match ::lrpar::RTParserBuilder::new(&grm, &stable)
             .recoverer(::lrpar::RecoveryKind::{recoverer})
-            .parse_actions(lexer, &actions) {{
-                (Some({actionskind}::{actionskindprefix}{ridx}(x)), y) => (Some(x), y),
-                (None, y) => (None, y),
+            .parse_actions(lexer, actions.as_slice(), {param_pat}) {{
+                (Some({actionskind}::{actionskindprefix}{ridx}(__x)), y, {param_pat}) => (Some(__x), y, {param_pat}),
+                (None, y, {param_pat}) => (None, y, {param_pat}),
                 _ => unreachable!()
         }}",
                     actionskind = ACTIONS_KIND,
                     actionskindprefix = ACTIONS_KIND_PREFIX,
                     ridx = usize::from(self.user_start_ridx(grm)),
                     recoverer = recoverer,
+                    param_pat = self.param_pattern(grm),
                 ));
             }
             YaccKind::Original(YaccOriginalActionKind::GenericParseTree) => {
                 outs.push_str(&format!(
                     "
-        ::lrpar::RTParserBuilder::new(&grm, &stable)
+        {{ let (__x, __y, _) = ::lrpar::RTParserBuilder::new(&grm, &stable)
```

Comment:
I'm about 98% certain that these variables are generated in a function which doesn't contain any user code. [I think that function has "bare" variables like `action` generated by `ctbuilder`.] If so, I think we can drop the `__` prefix.

Reply:
Yep, this highlights a missing test, as it wouldn't have compiled as is when running with `%parse_param` set in the yacc.